### PR TITLE
add script to build typechain from UI scripts

### PR DESCRIPTION
**Pull request by ChmilevFA**
* added new script to trigger `contracts` build and copy `typechain` to `UI` `src` folder
* moved `contracts` to yarn, to keep it consistent with `UI`

As @decimal-corn advised it might be better to have root scripts to manage such build dependencies. However I'm not sure that Spheron would handle it. Need to experiment first.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -26,8 +26,9 @@
     "web3modal": "^1.9.8"
   },
   "scripts": {
+    "contract-dependencies": "(cd ../contract && yarn && yarn run build)",
     "start": "react-scripts start",
-    "build": "react-scripts build",
+    "build": "yarn contract-dependencies && react-scripts build",
```
Comment (vlad324) [inline ui/package.json:31]: as I know `build` is only used for prod, to do local development we use `start`

when I was talking about the script I mean like just a shell script which will do compile of contracts, typechain generation and put types to ui folder

**Review by vlad324 — CHANGES_REQUESTED** [review]

File: ui/package.json (hunk 1 above)
Comment (vlad324) [reply ui/package.json:31]: just read your description :D
I would say I have doubts that Spheron will manage to compile contracts from an external folder as well

Comment (ChmilevFA) [reply ui/package.json:31]: Yes, just experimented with a few different option for Spheron and seems like it cannot make it.
I see three options here:
* Drop Spheron usage
* Find Spheron analogue which could make it
* Keep commiting generated typechain

For hackathon I would continue with last one to not drop Spheron. But if we decide to continue with the project after that - ofc need to refactor

Comment (vlad324) [reply ui/package.json:31]: yep, I think the last option for hackathon is fine

Comment (ChmilevFA) [reply ui/package.json]: Updated PR: kept scripts for manual trigger + used `yarn`

**Review by vlad324 — APPROVED** [review]

File: contract/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "hardhat-project",
   "scripts": {
     "test": "hardhat test",
-    "build": "hardhat compile --verbose && cp -R typechain-types ../ui/src"
+    "build": "hardhat compile --verbose && (rm -R ../ui/src/typechain-types || true) && cp -R typechain-types ../ui/src"
```
Comment (ChmilevFA) [inline contract/package.json:5]: just added remove before copy to not have outdated files

**Pull request merged**